Feature/contibutor service

IlyaBuryn opened the pull request:
Finish Contributor service.

IlyaBuryn committed:
Added data access and business logic for contributor service. Standard CRUD repository operations are separated into a separate base class.

IlyaBuryn committed:
Finish Data Access layer.

IlyaBuryn committed:
Finish Business Logic layer.

IlyaBuryn committed:
DTO, models and settings added to the shared library.

IlyaBuryn committed:
Added, moved or updated some classes in the shared library.

IlyaBuryn committed:
Finish API.

IlyaBuryn committed:
Fix some Aggregator service classes.

IlyaBuryn committed:
Added GET method to subscription controller.

KatsiarynaAnashkevich commented on Services/Aggregator/Aggregator.Api/Controllers/ComponentController.cs:
```diff
@@ -68,10 +72,9 @@ public async Task<IActionResult> GetComponentAvgMark([FromRoute] Guid componentI
         {
-            var response = await _componentService.GetComponentsAsPageAsync(pageNumber, pageSizeSettings.Value.PageSize, specificationFilter);
+            var response = await _componentService.GetComponentsAsPageAsync(pageNumber, _pageSizeSettings.PageSize, specificationFilter);
             return Ok(response);
```
empty line before return

KatsiarynaAnashkevich replied:
Everywhere

KatsiarynaAnashkevich commented on Services/Aggregator/Aggregator.Api/Program.cs:
```diff
@@ -25,19 +26,24 @@
+builder.Services.Configure<PageSizeSettings>(options =>
+    builder.Configuration.GetSection("PageSizeSettings").Bind(options));
+
+builder.Services.AddAuthentication(
```
Create extension method

KatsiarynaAnashkevich commented on Services/Aggregator/Aggregator.Api/Program.cs:
```diff
@@ -25,19 +26,24 @@
+        options.RequireHttpsMetadata = false;
     });
 
 builder.Services.AddAuthorization(options =>
```
And here

KatsiarynaAnashkevich commented on Services/Aggregator/Aggregator.DataAccess/Repositories/AggregatorEFRepository.cs:
```diff
@@ -1,130 +1,30 @@
-            return await Task.FromResult(query);
-        }
-
         public async Task<IQueryable<T?>> GetPageAsync(IQueryable<T?> set, int pageNumber, int pageSize)
```
Why do use return querable?

KatsiarynaAnashkevich commented on Services/Contributor/Contributor.Api/Program.cs:
```diff
@@ -0,0 +1,74 @@
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers(options =>
```
Extension method

KatsiarynaAnashkevich commented on Services/Contributor/Contributor.Api/Program.cs:
```diff
@@ -0,0 +1,74 @@
+
+builder.Services.AddAuthentication(
+IdentityServerAuthenticationDefaults.AuthenticationScheme)
+    .AddIdentityServerAuthentication(options =>
```
Extension method

KatsiarynaAnashkevich commented on Services/Contributor/Contributor.Api/Program.cs:
```diff
@@ -0,0 +1,74 @@
+        options.RequireHttpsMetadata = false;
+    });
+
+builder.Services.AddAuthorization(options =>
```
Extension method

KatsiarynaAnashkevich commented on Services/Contributor/Contributor.BusinessLogic/Services/ContributorService.cs:
```diff
@@ -0,0 +1,151 @@
+            var contributor = await _contributorRepo.GetOneEntityAsync(
+                includeProperties: x => x.ContributorExcellence,
+                expression: x => x.ContributorExcellence.Name == name);
+            if (contributor == null)
```
empty line

KatsiarynaAnashkevich commented on Services/Contributor/Contributor.DataAccess/Data/ContributorDbContext.cs:
```diff
@@ -0,0 +1,32 @@
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+            modelBuilder.ApplyConfiguration(new ChatMessageConfiguration());
```
You can apply them with one line of code

KatsiarynaAnashkevich commented on Services/HardwareHero.Services.Shared/DTOs/Validation/Aggregator/ComponentReviewValidator.cs:
```diff
@@ -1,7 +1,7 @@
-namespace HardwareHero.Services.Shared.DTOs.Validation
+namespace HardwareHero.Services.Shared.DTOs.Validation.Aggregator
 {
     [Validator(typeof(ComponentReviewDto))]
```
For what reason this attribute?

KatsiarynaAnashkevich commented on Services/HardwareHero.Services.Shared/DTOs/Validation/Contributor/ContributorExcellenceValidator.cs:
```diff
@@ -0,0 +1,25 @@
+
+namespace HardwareHero.Services.Shared.DTOs.Validation.Contributor
+{
+    [Validator(typeof(ContributorExcellenceDto))]
```
For what reason this attribute?

KatsiarynaAnashkevich commented on Services/HardwareHero.Services.Shared/DTOs/Validation/Contributor/ContributorValidator.cs:
```diff
@@ -0,0 +1,28 @@
+
+namespace HardwareHero.Services.Shared.DTOs.Validation.Contributor
+{
+    [Validator(typeof(ContributorDto))]
```
For what reason this attribute?

KatsiarynaAnashkevich commented on Services/HardwareHero.Services.Shared/DTOs/Validation/Contributor/SubscriptionInfoValidator.cs:
```diff
@@ -0,0 +1,19 @@
+
+namespace HardwareHero.Services.Shared.DTOs.Validation.Contributor
+{
+    [Validator(typeof(SubscriptionInfoDto))]
```
For what reason this attribute?

KatsiarynaAnashkevich commented:
One global comment: please, pay attention on your code style. Before return you have to leave one empty line)

IlyaBuryn committed:
Fixed empty lines, attributes, extensions and configuration.

Fixed empty lines. Removed attribute [Validator] from validators. Some code moved to the extension methods. ModelBuilder using `ApplyConfigurationsFromAssembly`.

IlyaBuryn committed:
Changed collection type return in `GetPageAsync` method.

Changed from IQueryable to IEnumerable.

IlyaBuryn committed:
Merge branch 'develop' into feature/contibutorService
